src/bpmncwpverify/builder/cwp_builder.py

```diff
@@ -0,0 +1,127 @@
+from typing import List
+from abc import ABC, abstractmethod
+from xml.etree.ElementTree import Element
+from bpmncwpverify.core.state import SymbolTable
+from bpmncwpverify.visitors.cwp_connectivity_visitor import CwpConnectivityVisitor
+from returns.result import Result, Success
+from bpmncwpverify.error import Error
+from bpmncwpverify.core.cwp import Cwp, CwpEdge, CwpState
+from bpmncwpverify.core.expr import ExpressionListener
+from returns.pipeline import is_successful
+from returns.functions import not_
+
+
+class CwpBuilder(ABC):
```

Comment:
Do we need `ABC` for the builder or even a base class for it? We can keep it, but I don't see us using it for anything other than creating the instance of the `Cwp`. The visitor for the `Cwp` is what we will use for generating different verification models, but the `Cwp` itself is our _intermediate_ representation for the specification.